### PR TITLE
fix: image/video picker upload issue from expo apps and getLocalAssetURI improvements

### PR DESCRIPTION
**Pull request by khushal87**
## 🎯 Goal

This PR focuses on fixing the images/video upload through expo apps and native CLI. This also streamlines the usage of `getLocalAssetUri` by optimizing its usage where we select an attachment rather than using it before actually uploading an attachment.

Moreover, the expo media library sends the `id` for every photo/video while the camera roll mostly deals with the `URI` of the attachment. Hence, the check should happen in the same way.

## 🎨 UI Changes

<details>
<summary>iOS</summary>

<table>
    <thead>
        <tr>
            <td>Before</td>
            <td>After</td>
        </tr>
    </thead>
    <tbody>
        <tr>
            <td>
                
            </td>
            <td>
                
            </td>
        </tr>
    </tbody>
</table>
</details>

<details>
<summary>Android</summary>

<table>
    <thead>
        <tr>
            <td>Before</td>
            <td>After</td>
        </tr>
    </thead>
    <tbody>
        <tr>
            <td>
                
            </td>
            <td>
                
            </td>
        </tr>
    </tbody>
</table>
</details>

## ☑️ Checklist

- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [ ] PR targets the `develop` branch
- [ ] Documentation is updated
- [ ] New code is tested in main example apps, including all possible scenarios
  - [ ] SampleApp iOS and Android
  - [ ] Expo iOS and Android

### CODE REVIEW

File: package/src/components/AttachmentPicker/components/AttachmentPickerItem.tsx [hunk 1]
```diff
@@ -139,17 +156,36 @@ const AttachmentImage: React.FC<AttachmentImageProps> = (props) => {
...
   const { uri } = asset;
 
+  const fetchUpdatedImages = async (images: Asset[]) => {
+    // For the case of Expo CLI where you need to fetch the file uri from file id. Here it is only done for iOS since for android the file.uri is fine.
+    const localAssetURI = Platform.OS === 'ios' && asset.id && (await getLocalAssetUri(asset.id));
+    const uri = localAssetURI || asset.uri || '';
+    if (numberOfUploads >= maxNumberOfFiles) {
+      Alert.alert('Maximum number of files reached');
+      return images;
+    }
```
Comment (santhoshvai) [inline package/src/components/AttachmentPicker/components/AttachmentPickerItem.tsx:166]: can we move this to line 188, basically inside else, seems irrerelvant here

File: package/src/components/AttachmentPicker/components/AttachmentPickerItem.tsx [hunk 2]
```diff
@@ -61,29 +64,42 @@ const AttachmentVideo: React.FC<AttachmentVideoProps> = (props) => {
 
   const size = vw(100) / (numberOfAttachmentPickerImageColumns || 3) - 2;
 
+  const fetchUpdatedFiles = async (files: File[]) => {
```
Comment (santhoshvai) [inline package/src/components/AttachmentPicker/components/AttachmentPickerItem.tsx:67]:
```suggestion
 /* Patches video files with uri and mimetype */
  const patchVideoFile = async (files: File[]) => {
```

**Review by santhoshvai — APPROVED** [review]

**Comment by khushal87** [comment]
@santhoshvai I have added your suggested changes. 🙂

**Comment by shahzaib803** [comment]
@santhoshvai Still same issue of image uploading in expo through camera
expo version 46.0.19
stream-chat-expo :5.11.2,

**Pull request merged**